cmd/soroban-cli/src/commands/tx/edit.rs

```diff
@@ -0,0 +1,186 @@
+        fs::remove_file(&path)?;
+
+        println!("{xdr}");
+
+        Ok(())
+    }
+}
+
+struct Editor {
+    cmd: String,
+    source: String,
+    args: Vec<String>,
+}
+
+fn tmp_file(contents: &str) -> Result<PathBuf, Error> {
+    let temp_dir = env::current_dir().unwrap_or(env::temp_dir());
+    let file_name = format!("stellar-xdr-{}.json", rand::random::<u64>());
+    let path = temp_dir.join(file_name);
+
+    let mut file = fs::File::create(&path)?;
+    file.write_all(contents.as_bytes())?;
+
+    Ok(path)
+}
+
+fn get_editor() -> Editor {
+    let (source, cmd) = env::var("STELLAR_EDITOR")
+        .map(|val| ("STELLAR_EDITOR", val))
+        .or_else(|_| env::var("EDITOR").map(|val| ("EDITOR", val)))
+        .or_else(|_| env::var("VISUAL").map(|val| ("VISUAL", val)))
+        .unwrap_or_else(|_| ("default", "vim".to_string()));
+
+    let parts: Vec<&str> = cmd.split_whitespace().collect();
+    let cmd = parts[0].to_string();
+    let args = &parts[1..]
+        .iter()
+        .map(|&s| s.to_string())
+        .collect::<Vec<String>>();
+
+    Editor {
+        source: source.to_string(),
+        cmd,
+        args: args.clone(),
+    }
+}
+
+fn open_editor(print: &Print, editor: &Editor, path: &PathBuf) -> Result<(), Error> {
+    print.infoln(format!(
+        "Opening editor with `{source}=\"{cmd}\"`...",
+        source = editor.source,
+        cmd = editor.cmd,
+    ));
+
+    let mut binding = process::Command::new(editor.cmd.clone());
+    let command = binding.args(editor.args.clone()).arg(path);
+
+    #[cfg(unix)]
+    {
+        use fs::File;
+        let tty = File::open("/dev/tty")?;
+        let tty_out = fs::OpenOptions::new().write(true).open("/dev/tty")?;
+        let tty_err = fs::OpenOptions::new().write(true).open("/dev/tty")?;
+
+        command
+            .stdin(tty)
+            .stdout(tty_out)
+            .stderr(tty_err)
+            .env("TERM", "xterm-256color");
+    }
+
+    let status = command.spawn()?.wait()?;
+
+    if status.success() {
+        Ok(())
+    } else {
+        Err(Error::EditorNonZeroStatus)
+    }
+}
+
+fn xdr_to_json<T>(xdr_string: &str) -> Result<String, Error>
+where
+    T: curr::ReadXdr + serde::Serialize,
+{
+    let version = env!("CARGO_PKG_VERSION");
+    let tx = T::from_xdr_base64(xdr_string, curr::Limits::none())?;
+    let mut schema: serde_json::Value = serde_json::to_value(tx).unwrap();
+    schema["$schema"] = json!(format!(
+        "https://github.com/stellar/stellar-cli/releases/download/v{version}/stellar-xdr-{version}.json"
+    ));
+    let json = serde_json::to_string_pretty(&schema)?;
+
+    Ok(json)
+}
+
+fn json_to_xdr<T>(json_string: &str) -> Result<String, Error>
+where
+    T: serde::de::DeserializeOwned + curr::WriteXdr,
+{
+    let mut schema: serde_json::Value = serde_json::from_str(json_string).unwrap();
+
+    if let Some(obj) = schema.as_object_mut() {
+        obj.remove("$schema");
+    }
```

Comment:
@leighmcculloch And this removes the `$schema`, before trying to convert it to xdr.

I tried a few things to use the `schema: serde_json::Value` directly when converting to xdr, but couldn't make it work. Is that possible at all?